cardano-testnet/src/Testnet/Defaults.hs

```diff
@@ -150,61 +151,54 @@ defaultConwayGenesis =
 
 -- | Configuration value that allows you to hardfork to any Cardano era
 -- at epoch 0.
-defaultYamlHardforkViaConfig :: AnyCardanoEra -> KeyMapAeson.KeyMap Aeson.Value
-defaultYamlHardforkViaConfig era =
+defaultYamlHardforkViaConfig :: ShelleyBasedEra era -> KeyMapAeson.KeyMap Aeson.Value
```

Comment:
Why are we dropping support for Byron era? If QA decides to use cardano-testnet in the future, we need a way to start Byron era nodes.

Reply:
@Jimbo4350> while introducing per-node options, I came across this TODO saying that we don't support Byron testnets anymore: https://github.com/IntersectMBO/cardano-node/blame/master/cardano-testnet/src/Parsers/Cardano.hs#L82

This is why I did this PR (also because it aligns with what we do on CLI and API).

Reply:
But is it even possible to start byron testnet at this very moment? I have no idea if anything < Babbage will work at this moment.
The big question is - should we support this?

Reply:
It should be possible. We should support this but perhaps in a similar approach to how it's done in the api. I.e we switch to `ShelleyBasedEra era` but we have code that can be specifically called to start a Byron testnet.